File: src/build_backend/post_requests/generate_questions/question_gen_request.py
# Import necessary libraries
import requests  # For making HTTP requests
import json  # For working with JSON data
from google.cloud import storage  # For interacting with Google Cloud Storage
from io import BytesIO  # For handling file content as bytes
import os  # For miscellaneous operating system functions
import PyPDF2  # For working with PDF files
import textract  # For extracting text from other file formats
from docx import Document  # For working with DOCX files
import tempfile  # For creating temporary files

# Cloud Run service URL
questions_api_url = "https://bitesize-question-gen-jsrdxhl2pa-ue.a.run.app/generate-questions"
extract_text_api_url = "https://bitesize-question-gen-jsrdxhl2pa-ue.a.run.app/extract-text"
#extract_text_api_url  = "http://10.250.146.64:8080/extract-text"
#api_url = 'http://10.250.146.64:80/generate-questions'
#api_url = "http://172.17.0.2:80/generate-questions"



# Make a GET request to the extract-text endpoint
get_response = requests.get(extract_text_api_url)

# Check if the GET request was successful
if get_response.status_code == 200:
    # Extract the text data from the response
    parsed_texts = get_response.json().get('document_text', [])

    # Define the request payload using the extracted text
    payload = {
        "text": "\n".join(parsed_texts),  # Combine text from all files
        "choice": "1"  # '1' for MCQ, '2' for short-answer (modify as needed)
    }

    # Send the POST request to the Cloud Run service
    post_response = requests.post(questions_api_url, json=payload)

    # Check the response status code
    if post_response.status_code == 200:
        # Print the JSON response
        response_data = post_response.json()
        print(json.dumps(response_data, indent=4))
    else:
        print(f"Error in POST request: {post_response.status_code} - {post_response.text}")
else:
    print(f"Error in GET request: {get_response.status_code} - {get_response.text}")

File: src/build_backend/post_requests/generate_questions/question_gen_request.js
function getExtractedTextAndSendRequest(userChoice) {
  // Step 1: Make a GET request to the Flask app to retrieve the extracted text
  fetch('https://bitesize-question-gen-jsrdxhl2pa-ue.a.run.app/extract-text')
    .then(response => {
      if (!response.ok) {
        throw new Error(`HTTP error! Status: ${response.status}`);
      }
      return response.json();
    })
    .then(data => {
      // Assuming the response has a "document_text" key with the extracted text
      const extractedText = data.document_text.join("\n");

      // Step 2: Construct the payload with the extracted text and user choice
      const payload = {
        text: extractedText,
        choice: userChoice, // This should be provided by the user in your app
      };

      // Step 3: Make a POST request to the Cloud Run service with the payload
      const url = "https://bitesize-question-gen-jsrdxhl2pa-ue.a.run.app/generate-questions";
      return fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });
    })
    .then(response => {
      if (!response.ok) {
        throw new Error(`HTTP error! Status: ${response.status}`);
      }
      return response.json();
    })
    .then(data => {
      console.log(data);
      // Handle the response from the Cloud Run service here
      // This is where you would update the UI with the questions and answers
    })
    .catch(error => {
      console.error('Error:', error);
      // Handle any errors that occurred during the fetch here
    });
}

// Example usage: this function can be called when the user submits their choice
// getExtractedTextAndSendRequest('2');
